### PR TITLE
Convert to dojo-interfaces

### LINKED ISSUE #77: Import base class interfaces from dojo/interfaces
Items to address:
- [ ] the base classes should be moved from `mixins` to `bases`
- [ ] the base classes should import their interfaces from `dojo-interfaces`
- [ ] these modules should also import other interfaces, like those for Observables and Handles from `dojo-interfaces`

### PR DESCRIPTION
**Pull request by agubler**
**Type:** feature

The following has been addressed in the PR:
- [x] There is a related issue
- [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
- [x] Unit or Functional tests are included in the PR

**Description:**

Convert to dojo-interfaces

Resolves #77

### CODE REVIEW

File: src/mixins/createEvented.ts [hunk 1]
```diff
@@ -141,43 +65,41 @@ function handlesArraytoHandle(handles: Handle[]): Handle {
...
+				}
+				else if (args.length === 1) { /* overload: on(listeners) */
+					const [ listenerMapArg ] = <[EventedListenersMap<any>]> args;
+					const handles = Object.keys(listenerMapArg).map((type) => this.on(type, listenerMapArg[type]));
 					return handlesArraytoHandle(handles);
 				}
-				else {
-					return on(listenerMap, type, resolveListener(listeners));
+				else { /* unexpected signature */
+					throw new TypeError('Invalid arguments');
```
Comment (kitsonk) [inline src/mixins/createEvented.ts:97]: I suspect there is no test case that covers this.

**Review by kitsonk — COMMENTED** [review]
So what is the plan with migrating to `src/bases`?  Would it be then that `src/bases/Stateful` would totally eliminate the "local" interfaces and take everything from `dojo-interfaces`?  Would `src/bases/Destroyable` and `src/bases/Evented` be copy pastas of what is here?

**Comment by agubler** [comment]
@kitsonk yes we would copy the supporting classes (`createEvented`/`createDestroyable`)to `src/bases` and the `bases/createStateful` would indeed remove all local interfaces

File: src/mixins/createEvented.ts (hunk 1 above)
Comment (kitsonk) [reply src/mixins/createEvented.ts:97]: Nope there is... :-D

**Review by kitsonk — APPROVED** [review]
👍

**Pull request merged**